result/src/main/kotlin/com/github/kittinunf/result/Result.kt

```diff
@@ -28,30 +28,57 @@ fun <V: Any?, E: Exception> Result<V, E>.getOrNull(): V? {
     }
 }
 
-inline fun <V : Any?, U : Any?, E : Exception> Result<V, E>.map(transform: (V) -> U): Result<U, E> = try {
+inline infix fun <V : Any?, E : Exception> Result<V, E>.getExceptionOrElse(fallback: (V) -> E): E {
```

Comment:
Just a question though, in what scenario `getExceptionOrElse` makes sense to use? I personally have never feel that I want to just to get the exception from `Result`?

Reply:
Agreed there are not that many cases, but we do have a few, mostly for testing however. Also happy to remove this one if you feel it is not needed enough.

Reply:
Yeah, I think at least I would love to keep the API surface slim. If it is not common enough, do you think it is worthwhile to keep it from the library and have it as a specific extension in client's repo?

Reply:
Yeah I think that is fair enough.